### PR TITLE
podman-remote pause|unpause

### PR DESCRIPTION
**Pull request by baude**
Add the ability to pause and unpause containers with the remote client.

Also turned on the pause tests!

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: test/e2e/pause_test.go [hunk 1]
```diff
@@ -1,14 +1,11 @@
...
 package integration
 
 import (
 	"fmt"
-	"os"
-
 	. "github.com/containers/libpod/test/utils"
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	"os"
```
Comment (rhatdan) [inline test/e2e/pause_test.go]: Misplaced

**Comment by TomSweeneyRedHat** [comment]
Changes LGTM but the tests aren't hip.

File: pkg/adapter/containers.go [hunk 2]
```diff
@@ -607,3 +607,90 @@ func (r *LocalRuntime) Start(ctx context.Context, c *cliconfig.StartValues, sigP
...
+	maxWorkers := shared.DefaultPoolSize("pause")
+	if cli.GlobalIsSet("max-workers") {
+		maxWorkers = cli.GlobalFlags.MaxWorks
+	}
+	logrus.Debugf("Setting maximum rm workers to %d", maxWorkers)
+
+	if cli.All {
+		ctrs, err = r.GetRunningContainers()
+	} else {
+		ctrs, err = shortcuts.GetContainersByContext(false, false, cli.InputArgs, r.Runtime)
```
Comment (mheon) [inline pkg/adapter/containers.go]: Isn't all a bool in this function? Don't need the if check, I think

File: pkg/adapter/containers.go [hunk 3]
```diff
@@ -607,3 +607,90 @@ func (r *LocalRuntime) Start(ctx context.Context, c *cliconfig.StartValues, sigP
...
+
+	pool := shared.NewPool("pause", maxWorkers, len(ctrs))
+	for _, c := range ctrs {
+		ctr := c
+		pool.Add(shared.Job{
+			ID: ctr.ID(),
+			Fn: func() error {
+				err := ctr.Pause()
+				if err != nil {
+					logrus.Debugf("Failed to pause container %s: %s", ctr.ID(), err.Error())
```
Comment (mheon) [inline pkg/adapter/containers.go]: Do we need this if we're going to print errors anyways?

File: pkg/adapter/containers.go [hunk 4]
```diff
@@ -607,3 +607,90 @@ func (r *LocalRuntime) Start(ctx context.Context, c *cliconfig.StartValues, sigP
...
+
+	pool := shared.NewPool("pause", maxWorkers, len(ctrs))
+	for _, c := range ctrs {
+		ctr := c
+		pool.Add(shared.Job{
+			ID: ctr.ID(),
+			Fn: func() error {
+				err := ctr.Unpause()
+				if err != nil {
+					logrus.Debugf("Failed to unpause container %s: %s", ctr.ID(), err.Error())
```
Comment (mheon) [inline pkg/adapter/containers.go]: Again - do we need the logrus here?

File: pkg/adapter/containers.go (hunk 2 above)
Comment (mheon) [reply pkg/adapter/containers.go]: Actually, nevermind - I think we're missing some logic on the `cli.All` if - we should be grabbing only running containers

File: test/utils/utils.go [hunk 5]
```diff
@@ -4,6 +4,7 @@ import (
 	"bufio"
 	"encoding/json"
 	"fmt"
+	"github.com/containers/libpod/pkg/rootless"
```
Comment (mheon) [inline test/utils/utils.go]: Input ordering

**Comment by mheon** [comment]
I think there's a logic error in local pause code, plus a few more nits, LGTM once addressed

File: pkg/adapter/containers.go (hunk 3 above)
Comment (baude) [reply pkg/adapter/containers.go]: it is a straight copy from @jwhonce 's code.

File: pkg/adapter/containers.go (hunk 4 above)
Comment (baude) [reply pkg/adapter/containers.go]: it is a straight copy from @jwhonce 's code.

File: pkg/adapter/containers.go (hunk 2 above)
Comment (baude) [reply pkg/adapter/containers.go]: The man page for pause says all should pause all running containers.  So the way I read this is 626 says if all, get all running containers .... seems correct to me?

Comment (mheon) [reply pkg/adapter/containers.go]: Ah, I see

**Pull request merged**